src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -3170,18 +3170,23 @@ private static Expression CompareWithZero(DynamicMetaObject target, Func<Express
             return comparer(target.Expression.Cast(target.LimitType), ExpressionCache.Constant(0).Cast(target.LimitType)).Cast(typeof(object));
```

Comment:
@vexx32, your last commit had 12 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Null Representatives.Numeric Comparisons.{ $null } should be considered equal to 0
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 103
103:             0 -eq $Value | Should -BeTrue
```
Null Representatives.Numeric Comparisons.{ [DBNull]::Value } should be considered equal to 0
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 103
103:             0 -eq $Value | Should -BeTrue
```
Null Representatives.Numeric Comparisons.{ [NullString]::Value } should be considered equal to 0
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 103
103:             0 -eq $Value | Should -BeTrue
```
Null Representatives.Numeric Comparisons.{ [AutomationNull]::Value } should be considered equal to 0
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 103
103:             0 -eq $Value | Should -BeTrue
```
Null Representatives.Numeric Comparisons.{ $null } should be greater than -1
```powershell
Cannot compare " $null " because it is not IComparable.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 115
115:             $Value -gt -1 | Should -BeTrue

```